ruby_on_rails/testing/model_testing.md

```diff
@@ -0,0 +1,136 @@
+
+## Setting the test
+
+The first tool that we are going to use to set up the data is [factory bot](https://github.com/thoughtbot/factory_bot). Factory bot is a gem that you can use to set up test data and create necessary data for your tests. Adding Factory bot is similar to adding other testing gems. Add the following line to the to the following section in your Gemfile.
+
+~~~
+
+group :development, :test do
+
+  gem "factory_bot_rails"  #Add this line.
+
+end
+
+~~~
+
+Then in your terminal enter `bundle install` to the terminal where it will install the gem. Now we need to move into setting up our test configuration to utilize Factory bot. First were going to navigate to `spec/support/factory_bot.rb` and add the following configuration. 
+
+~~~
+
+RSpec.configure do |config|
+  config.include FactoryBot::Syntax::Methods
+end
+
+~~~
+
+Make sure to require this file in `rails_helper.rb`. We will briefly cover what a factory will look like and some things to keep in mind when creating your data. While you can require the file by adding `require support/factory_bot` to the top of the file. Rails also provides this in the commented out line `Dir[Rails.root.join("spec/support/**/*.rb")].each { |f| require f }` which enables you to load all of the support files when booting up. This is useful for not having to manually add all of the support files and avoids issues if you forget to add a file for example. The last thing you might be wondering is how the test database clears itself in between it's tests. Rail's does this automatically for you, however if Rail's is not doing it for you make sure your following the instructions in the rails_helper.rb file and make sure to ask in the [Discord](https://discord.com/channels/505093832157691914/690591236922409012)! There is also a alternative option in the additional resources section if need be for database cleaning.
+
+## Creating a factory
+
+Below is an example of a user factory. Notice that the factory will be a User class based off of the name of your factory. You can also define as many attributes as you want and it can be useful to focus on adding only the attributes your testing and adding them as need be.
+
+~~~
+
+# This will guess the User class
+FactoryBot.define do
+  factory :user do
+    first_name { "John" }
+    last_name  { "Doe" }
+    admin { false }
+  end
+end
+
+~~~
+
+When creating a factory bot record you have the choice of using the `create(:user)` which returns a saved instance of the object from your database or `build(:user)` which returns an unsaved instance of an object. You might be wondering how we avoid objects hanging around in the database. Thankfully RSpec includes the ability to clear the database in between tests. However if you do find that RSpec's database clearing isn't fitting your needs here you can check the additional resources for alternatives.
```

Comment:
I don't think we need to keep mentioning cleaning up the database. I believe it is detracting from the lesson content.